samza-core/src/main/java/org/apache/samza/table/TableManager.java

```diff
@@ -125,6 +130,14 @@ private void addTable(TableSpec tableSpec) {
     tables.put(tableSpec.getId(), ctx);
   }
 
+  /**
+   * Set TaskContext for table instance initialization
+   * @param taskContext
+   */
+  public void setTaskContext(TaskContext taskContext) {
```

Comment:
Good point. Will make the change.

Reply:
Talked to @weisong44 offline, we think it's better to leave the init in SamzaContainer because it instantiates the TableManager and passes it to TaskInstance. From this standpoint, container should own the initialization of TM. For now, I'll keep it as is. Let me know if you are still concerned. 

Also, both setTaskContext and initLocalTables have been replaced with a unified init() method.

Reply:
>> Talked to @weisong44 offline, we think it's better to leave the init in SamzaContainer because it instantiates the TableManager and passes it to TaskInstance. From this standpoint, container should own the initialization of TM. For now, I'll keep it as is. Let me know if you are still concerned.

1. I did have a follow-up discussion on this @weisong44 today. To keep the initialization flow consistent, I still think it makes sense to consolidate initializations in one place. For instance, the `TaskStorageManager` is initialized inside the `TaskInstance` by invoking `taskInstance.startStores`.  I'd think that the `TableManager` should be initialized in an analogous manner.

2. In addition to not leaking the `taskContext` object outside of the `taskInstance`, it will also 
ensure that anything that requires the `taskContext` to be initialized within the scope of the `TaskInstance`

3. Unrelated to this PR, but I would also evaluate if we need both init() and a start() for the `TableManager`. If there's no compelling case, it'd be simpler to combine the two. FWIW, the other managers like `StorageManager` , `CheckpointManager`, `SystemAdmin` have a single method. 

>> Also, both setTaskContext and initLocalTables have been replaced with a unified init() method.

Nice, love it!